marimo/_plugins/ui/_impl/chat/llm.py

```diff
@@ -307,3 +310,86 @@ def __call__(
 
         content = response.text
         return content or ""
+
+
+class groq(ChatModel):
+    """
+    Groq ChatModel
+
+    **Args:**
+
+    - model: The model to use.
+        Can be found on the [Groq models page](https://console.groq.com/docs/models)
+    - system_message: The system message to use
+    - api_key: The API key to use.
+        If not provided, the API key will be retrieved
+        from the GROQ_API_KEY environment variable or the user's config.
+    - base_url: The base URL to use
+    """
+
+    def __init__(
+        self,
+        model: str,
+        *,
+        system_message: str = DEFAULT_SYSTEM_MESSAGE,
+        api_key: Optional[str] = None,
+        base_url: Optional[str] = None,
+    ):
+        self.model = model
+        self.system_message = system_message
+        self.api_key = api_key
+        self.base_url = base_url
+
+    @property
+    def _require_api_key(self) -> str:
+        # If the api key is provided, use it
+        if self.api_key is not None:
+            return self.api_key
+
+        # Then check the environment variable
+        env_key = os.environ.get("GROQ_API_KEY")
+        if env_key is not None:
+            return env_key
+
+        # Then check the user's config
+        try:
+            from marimo._runtime.context.types import get_context
+
+            api_key = get_context().user_config["ai"]["groq"]["api_key"]
```

Comment:
technically we don't have this config, but we can add it later if we want

Reply:
Right; it would be more ideal if I added it as a chat completion feature right? From the file, I'm guessing it involves mainly that. Should it be commented or even removed now?

Reply:
Commented out with a todo is great, thank you!